plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/MavenPath.java

```diff
@@ -335,6 +361,69 @@ public MavenPath signature(final SignatureType signatureType) {
     );
   }
 
+  /**
+   * Returns path pointing to given extension and optionally classifier within this same GAV. Only usable for artifact
+   * paths, those having non-null {@link #getCoordinates()}.
+   */
+  @Nonnull
+  public MavenPath locate(final String extension, @Nullable final String classifier) {
+    checkNotNull(extension);
+    checkArgument(coordinates != null, "Only artifact paths may locate: %s", this);
+
+    MavenPath origin = this;
+    if (isHash()) {
+      origin = origin.subordinateOf();
+    }
+    if (isSignature()) {
+      origin = origin.subordinateOf();
+    }
+    Coordinates coordinates = new Coordinates(
+        origin.coordinates.isSnapshot(),
+        origin.coordinates.getGroupId(),
+        origin.coordinates.getArtifactId(),
+        origin.coordinates.getVersion(),
+        origin.coordinates.getTimestamp(),
+        origin.coordinates.getBuildNumber(),
+        origin.coordinates.getBaseVersion(),
+        classifier,
+        extension,
+        null
```

Comment:
SignatureType is basically inherited from Maven, that left room for other signature types implementation (other than GPG). In short, Maven implements only GPG one, but nothing forbids you to add/roll your own (ie. in an extension).